Framework/API/inc/MantidAPI/ComponentInfo.h

```diff
@@ -58,6 +59,8 @@ class MANTID_API_DLL ComponentInfo {
                 std::vector<Mantid::Geometry::IComponent *> &&componentIds);
   std::vector<size_t> detectorIndices(size_t componentIndex) const;
   size_t size() const;
+  Eigen::Vector3d position(const size_t componentIndex) const;
+  Eigen::Quaterniond rotation(const size_t componentIndex) const;
```

Comment:
Can we use `Eigen` here or do we need to convert to the types in `Kernel`, as in `DetectorInfo?